lib/elixir/lib/integer.ex

```diff
@@ -400,17 +400,23 @@ defmodule Integer do
       iex> Integer.gcd(10, 0)
       10
 
+      iex> Integer.gcd(7, 7)
+      7
+
       iex> Integer.gcd(0, 0)
       0
 
   """
+  @spec gcd(0, 0) :: 0
+  @spec gcd(int, int) :: int when int: pos_integer
```

Comment:
> Wouldn't that mean equality? If so, either the docs are wrong (or extremely confusing) or it does enforce equality.

No, **you have to think about subtyping**. You are thinking `number()` in types means 0, 1, 2, etc. But `number` in types mean `integer` or `float`.